### PR TITLE
Add datetime type to created/lastmodified date, add additional contai…

### PR DESCRIPTION
**Pull request by whikloj**
…ner types

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3223

This PR does not fulfill all of the above ticket.

# What does this Pull Request do?
Adds a bunch of other `rdf:type` to a Container. Also adds the appropriate xsd:datetime type to the created date and last modified date server managed triples.

# How should this be tested?

Build and create a RDF source, then GET it.
Before you will get the interaction model (ie. `ldp:DirectContainer`) and `ldp:RdfSource` types
Now you will also `ldp:Container`, `fedora:Container` and `fedora:Resource` types.
Also the `fedora:created` and `fedora:lastmodified` will now have the `xsd:dateTime` type 

# Interested parties
@fcrepo4/committers

### CODE REVIEW

**Comment by whikloj** [comment]
Amended to fix the response when trying to add a resource inside a binary.

**Comment by whikloj** [comment]
I think this might also resolve https://jira.lyrasis.org/browse/FCREPO-3168

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImpl.java [hunk 1]
```diff
@@ -194,8 +195,8 @@ private void checkParent(final String txId, final PersistentStorageSession pSess
             final ResourceHeaders parent;
             try {
                 // Make sure the parent exists.
+                // TODO: object existence can be from the index, but we don't have interaction model. Should we add it?
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImpl.java]: That seems like a reasonable optimization.

Comment (whikloj) [reply fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImpl.java]: https://jira.lyrasis.org/browse/FCREPO-3258

**Pull request merged**